R/hazardFunctions.R

```diff
@@ -0,0 +1,21 @@
+#' OS Hazard Function from Constant Transition Hazards
+#'
+#' @param t (`numeric`)\cr study time-points.
+#' @param h01 (positive `number`)\cr transition hazard for 0 to 1 transition.
+#' @param h02 (positive `number`)\cr transition hazard for 0 to 2 transition.
+#' @param h12 (positive `number`)\cr transition hazard for 1 to 2 transition.
+#'
+#' @return This returns the value of the OS hazard function at time t.
+#' @export
+#'
+#' @examples
+#' ExpHazOS(c(1:5), 0.2, 1.1, 0.8)
+ExpHazOS <- function(t, h01, h02, h12) {
+  assert_numeric(t, lower = 0, any.missing = FALSE)
+  assert_positive_number(h01, zero_ok = TRUE)
```

Comment:
Are you sure zero is ok?

Reply:
No :) Tbh I just used https://github.com/insightsengineering/simIDM/blob/016fe296ad37344c390c3db2ed25441f5b8137bd/R/survivalFunctions.R#L11C10-L11C10 for reference. I figured since S(t) and f(t) are defined for t =0, so should h(t). What do you think?

Reply:
`t == 0` should be ok, I agree - but I don't think the hazard itself `h(t)` should be allowed to be zero? otherwise in the calculation later we might get problems?

Reply:
I see, I changed it now